server/src/test/java/org/opensearch/indices/IndicesRequestCacheTests.java

```diff
@@ -535,6 +537,80 @@ public void testSerializationDeserializationOfCacheKey() throws Exception {
 
     }
 
+    public void testWithOnHeapCacheTypeSetting() throws Exception {
+        IndicesService indicesService = getInstanceFromNode(IndicesService.class);
+        IndexShard indexShard = createIndex("test").getShard(0);
+        Settings settings = Settings.builder().put(INDICES_REQUEST_CACHE_TYPE.getKey(), CacheType.ON_HEAP).build();
+        IndicesRequestCache cache = new IndicesRequestCache(settings, (shardId -> {
+            IndexService indexService = null;
+            try {
+                indexService = indicesService.indexServiceSafe(shardId.getIndex());
+            } catch (IndexNotFoundException ex) {
+                return Optional.empty();
+            }
+            return Optional.of(new IndicesService.IndexShardCacheEntity(indexService.getShard(shardId.id())));
+        }));
+        Directory dir = newDirectory();
+        IndexWriter writer = new IndexWriter(dir, newIndexWriterConfig());
+
+        writer.addDocument(newDoc(0, "foo"));
+        DirectoryReader reader = OpenSearchDirectoryReader.wrap(DirectoryReader.open(writer), new ShardId("foo", "bar", 1));
+        TermQueryBuilder termQuery = new TermQueryBuilder("id", "0");
+        BytesReference termBytes = XContentHelper.toXContent(termQuery, MediaTypeRegistry.JSON, false);
+
+        // initial cache
+        IndicesService.IndexShardCacheEntity entity = new IndicesService.IndexShardCacheEntity(indexShard);
+        Loader loader = new Loader(reader, 0);
+        BytesReference value = cache.getOrCompute(entity, loader, reader, termBytes);
+        assertEquals("foo", value.streamInput().readString());
+        ShardRequestCache requestCacheStats = entity.stats();
+        assertEquals(0, requestCacheStats.stats().getHitCount());
```

Comment:
Do we have tests anywhere for hit count incremented, evictions incremented, etc? Or plan this for the future?